src/charm.py

```diff
@@ -261,38 +261,71 @@ def _get_gateway_apis_kubernetes_resource_manager(self):
             logger=LOGGER,
         )
 
-    def _workload_tracing_config(self) -> Dict[str, str]:
-        """Set workload tracing configuration."""
-        tracing_config = {}
-        if not self.workload_tracing.is_ready():
-            return tracing_config
-
-        workload_tracing_endpoint = self.workload_tracing.get_endpoint("otlp_grpc")
-        if not workload_tracing_endpoint:
-            return tracing_config
-
-        tracing_config["meshConfig.enableTracing"] = "true"
-        tracing_config["meshConfig.extensionProviders[0].name"] = "otel-tracing"
-        tracing_config["meshConfig.extensionProviders[0].opentelemetry.port"] = urlparse(
-            f"//{workload_tracing_endpoint}"
-        ).port
-        tracing_config["meshConfig.extensionProviders[0].opentelemetry.service"] = urlparse(
-            f"//{workload_tracing_endpoint}"
-        ).hostname
-        tracing_config["meshConfig.defaultProviders.tracing[0]"] = "otel-tracing"
-        tracing_config["meshConfig.defaultConfig.tracing.sampling"] = 100.0
-
-        return tracing_config
+    def _workload_tracing_provider(self) -> Tuple[List[Any], Dict[str, Any]]:
+        """Return a tuple with a list containing the tracing provider and global tracing settings."""
+        providers = []
+        global_config = {}
+        if self.workload_tracing.is_ready():
+            workload_tracing_endpoint = self.workload_tracing.get_endpoint("otlp_grpc")
+            if workload_tracing_endpoint:
+                parsed = urlparse(f"//{workload_tracing_endpoint}")
+                providers.append(
+                    {
+                        "name": "otel-tracing",
+                        "opentelemetry": {
+                            "port": str(parsed.port),
+                            "service": parsed.hostname,
+                        },
+                    }
+                )
+                global_config = {
+                    "meshConfig.enableTracing": "true",
+                    "meshConfig.defaultProviders.tracing": "otel-tracing",
+                    "meshConfig.defaultConfig.tracing.sampling": 100.0,
+                }
+        return providers, global_config
+
+    def _flatten_config(self, value: Any, prefix: str = "") -> Dict[str, Any]:
+        """Recursively flatten a nested dictionary or list into a dictionary of key/value pairs."""
+        flat: Dict[str, Any] = {}
+        if isinstance(value, dict):
+            for k, v in value.items():
+                new_prefix = f"{prefix}.{k}" if prefix else k
+                flat.update(self._flatten_config(v, new_prefix))
+        elif isinstance(value, list):
```

Comment:
Probably we can include tuples here too
```suggestion
        elif isinstance(value, (list, tuple)):  # I think this is the syntax?
```